crates/iota-light-client/src/config.rs

```diff
@@ -0,0 +1,173 @@
+            graphql_url: Some("http://localhost:9125".to_string()),
+            checkpoints_dir: std::env::current_dir()
+                .expect("error getting current directory")
+                .join("checkpoints_localnet"),
+            sync_before_check: false,
+            object_store_url: None,
+            archive_store_config: None,
+        }
+    }
+}
+
+impl Config {
+    pub fn load<P: AsRef<Path>>(path: P) -> Result<Self> {
+        let file = fs::File::open(path)?;
+        let config: Config = serde_yaml::from_reader(file)?;
+        config.validate()?;
+        Ok(config)
+    }
+
+    pub fn validate(&self) -> Result<()> {
+        Url::parse(&self.rpc_url).map_err(|_| anyhow!("Invalid full node URL"))?;
+        if !self.checkpoints_dir.is_dir() {
+            bail!("Checkpoint directory does not exist");
+        }
+        if let Some(url) = &self.object_store_url {
+            Url::parse(url).map_err(|_| anyhow!("Invalid object store URL"))?;
+        }
+        if let Some(url) = &self.graphql_url {
+            Url::parse(url).map_err(|_| anyhow!("Invalid GraphQL URL"))?;
+        }
+        if !self.checkpoints_list_file_path().is_file() {
+            bail!(
+                "Sync file is missing at {}",
+                self.checkpoints_list_file_path().display()
+            );
+        }
+        if !self.genesis_blob_file_path().is_file() {
+            bail!(
+                "Genesis file is missing at {}",
+                self.genesis_blob_file_path().display()
+            );
+        }
+        Ok(())
+    }
+
+    pub fn checkpoints_list_file_path(&self) -> PathBuf {
+        self.checkpoints_dir.join(CHECKPOINTS_FILE_NAME)
+    }
+
+    pub fn genesis_blob_file_path(&self) -> PathBuf {
+        self.checkpoints_dir.join(GENESIS_FILE_NAME)
+    }
+
+    pub fn full_checkpoint_file_path<'a>(
+        &self,
+        seq: u64,
+        custom_path: impl Into<Option<&'a str>>,
+    ) -> PathBuf {
+        let mut path = self.checkpoints_dir.clone();
+        if let Some(custom) = custom_path.into() {
+            path.push(custom);
+        }
+        path.push(format!("{seq}.chk"));
+        path
+    }
+
+    pub fn checkpoint_summary_file_path<'a>(
+        &self,
+        seq: u64,
+        custom_path: impl Into<Option<&'a str>>,
+    ) -> PathBuf {
+        let mut path = self.checkpoints_dir.clone();
+        if let Some(custom) = custom_path.into() {
+            path.push(custom);
+        }
+        path.push(format!("{seq}.sum"));
+        path
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use iota_config::object_storage_config::ObjectStoreType;
+    use tempfile::TempDir;
+
+    use super::*;
+
+    fn create_test_config() -> (Config, TempDir) {
+        let temp_dir = TempDir::new().unwrap();
+        std::fs::File::create(temp_dir.path().join("genesis.blob")).unwrap();
+        std::fs::File::create(temp_dir.path().join("checkpoints.yaml")).unwrap();
```

Comment:
These files aren't needed here?

Reply:
They are, but I forgot to call `config.validate`. Fixed.